eva_cttv_pipeline/evidence_string_generation/evidence_strings.py

```diff
@@ -28,50 +32,37 @@ def get_cttv_variant_type(clinvar_record_measure):
-
-    Example: 'Benign/Likely benign, risk_factor' → ['benign', 'likely benign', 'risk factor'].
-    """
-    return sorted(re.split('/|, ', clin_sig.lower().replace('_', ' ')))
-
-
 class CTTVEvidenceString(dict):
-    """
-    Base evidence string class. Holds variables and methods common between somatic and genetic
-    evidence strings.
-    Subclass of dict to use indexing.
-    """
-
-    def __init__(self, a_dictionary, clinvar_record=None, ref_list=None,
-                 ensembl_gene_id=None, report=None, trait=None):
-        super().__init__(a_dictionary)
+    """Base evidence string class. Holds variables and methods common between somatic and genetic evidence strings.
+    Subclass of dict to use indexing."""
 
-        if ensembl_gene_id:
-            self.add_unique_association_field('gene', ensembl_gene_id)
-        if clinvar_record:
-            self.add_unique_association_field('clinvarAccession', clinvar_record.accession)
+    def __init__(self, a_dictionary, clinvar_record, clinvar_trait, ontology_id, ontology_label, ensembl_gene_id):
+        super().__init__(a_dictionary)
 
-        if ensembl_gene_id:
-            ensembl_gene_id_uri = get_ensembl_gene_id_uri(ensembl_gene_id)
-            self.set_target(ensembl_gene_id_uri)
+        # Add unique association fields. When considered together, a tuple of them is intended to be unique to each
+        # evidence string and to be able to serve as a key for querying.
+        self.add_unique_association_field('clinvarAccession', clinvar_record.accession)
+        self.add_unique_association_field('gene', ensembl_gene_id)
+        self.add_unique_association_field('phenotype', ontology_id)
 
-        if ref_list and len(ref_list) > 0:
-            self.top_level_literature = ref_list
+        # Populate target information
+        self.set_target(get_ensembl_gene_id_uri(ensembl_gene_id))
 
-        self.disease_id = trait.ontology_id
-        self.add_unique_association_field('phenotype', trait.ontology_id)
+        # Populate disease information
+        self.disease_id = ontology_id
+        self.disease_name = ontology_label
+        self.disease_source_name = clinvar_trait.name
 
-        if trait.clinvar_name:
-            self.disease_source_name = trait.clinvar_name
+        # Populate literature reference information
+        ref_list = list(set(clinvar_trait.pubmed_refs +             # Trait-specific references
+                            clinvar_record.measure.pubmed_refs +    # Variant-specific references
+                            clinvar_record.observed_pubmed_refs))   # "ObservedIn" references
+        self.full_ref_list = sorted(clinvar_xml_utils.pubmed_refs_to_urls(ref_list))
+        if self.full_ref_list:
+            self.top_level_literature = self.full_ref_list
 
-        if trait.ontology_label:
-            self.disease_name = trait.ontology_label
+        if len(self.full_ref_list) != len(set(self.full_ref_list)):
+            print('WTF')
```

Comment:
That does not seem to be an appropriate error message.

Reply:
Lol... May be state something like "Duplicate entries in the resolved URLs for Pubmed references"... Also, shouldn't this check be done before assignment to **top_level_literature**

Reply:
Another remnant of intense debugging which I intended to fix before submitting a PR and just forgot... Sorry :sweat:

Now fixed: https://github.com/EBIvariation/eva-opentargets/pull/167/commits/7b455c2a2b53219920b7c691eabe2c3d3971ff2c